### DISCUSSION TITLE
using gix on a rust application

### DISCUSSION THREAD

**Discussion by omer-shtivi** [GitoxideLabs/gitoxide#1381] [Q&A] [answered]
Hello,
We have rust application which need to use git client at the backend.
It needs to clone a remote repo, update the files in the repo, commit and push.
We tried to use gix, but seems like it is too low level for us, and some APIs are not that well documented.
Is there a way to work on high level abstraction (e.g. pull/clone/ commit/push)? or are we missing something?

And BTW
what is progress? `gix::progress`

**Comment by Byron** [comment, marked as answer]
[Many parts](https://github.com/Byron/gitoxide?tab=readme-ov-file#feature-discovery) of the workflow you describe aren't implemented yet, and I'd think you'd be better of with `git2` at the moment.

Using the `git` binary might still be easiest for many tasks especially in an environment that is under your control. For reading data directly from a Git repository, I'd recommend using `gitoxide` though, I don't think you will find anything better or faster ([NoseyParker](https://github.com/praetorian-inc/noseyparker) is a good example for that).

**Reply by omer-shtivi** [reply]
Thanks for your reply 
Is there a future plan to support those functionality ? 
We really prefer to use native rust

**Reply by Byron** [reply]
Yes, these checkboxes will definitely be ticked. It's just a question of when. My recommendation here stands - use `gitoxide` for what it can do today, which should be a lot related to interacting with existing repositories - depending on what the app does it's fine to have a bare clone and do everything else on the Git data directly - if that's multi-threaded it can be faster than if it was on disk even, which at some point becomes IO bound.

**Reply by omer-shtivi** [reply]
Thanks,
I will keep an eye out for releases, if I will have spare time I will contribute
